neqo-http3/src/settings.rs

```diff
@@ -384,4 +384,78 @@ mod tests {
             Err(Error::NotEnoughData)
         );
     }
+
+    #[test]
+    fn datagram_settings() {
+        for setting in [SETTINGS_H3_DATAGRAM, SETTINGS_H3_DATAGRAM_DRAFT04] {
+            // Valid value accepted.
+            let mut enc = Encoder::new();
+            enc.encode_varint(setting).encode_varint(1u64);
+            let mut s = HSettings::new(&[]);
+            s.decode_frame_contents(&mut enc.as_decoder()).unwrap();
+            assert_eq!(s.get(HSettingType::EnableH3Datagram), 1);
+
+            // Invalid value rejected.
+            enc = Encoder::new();
+            enc.encode_varint(setting).encode_varint(2u64);
+            let mut s = HSettings::new(&[]);
+            assert_eq!(
+                s.decode_frame_contents(&mut enc.as_decoder()),
+                Err(Error::HttpSettings)
+            );
+        }
+
+        // Duplicate: first wins.
+        let mut enc = Encoder::new();
+        enc.encode_varint(SETTINGS_H3_DATAGRAM).encode_varint(1u64);
+        enc.encode_varint(SETTINGS_H3_DATAGRAM_DRAFT04)
+            .encode_varint(0u64);
+        let mut s = HSettings::new(&[]);
+        s.decode_frame_contents(&mut enc.as_decoder()).unwrap();
+        assert_eq!(s.get(HSettingType::EnableH3Datagram), 1);
```

Comment:
So, if this were to put the DRAFT04 thing first, would that win?  Why not test that?